### PR TITLE
Add support for complex destinations

### PR DESCRIPTION
**Pull request by lcarva**
Signed-off-by: Luiz Carvalho <lucarval@redhat.com>

### CODE REVIEW

File: message_tagging_service/tagging_service.py [hunk 1]
```diff
@@ -259,17 +261,46 @@ def match(self, modulemd):
...
+        # ability to support this use case by generating permutations of each
+        # possible combination and applying each of these permutations to generate
+        # the destination tag name.
+
+        # Aggregate the different values for each regex group.
+        replacements = {}
+        for group_dict in self._regex_group_dicts:
+            for group, value in group_dict.items():
+                replacements.setdefault(group, [])
+                replacements[group].append(value)
```
Comment (tkdchen) [inline message_tagging_service/tagging_service.py]: Why need to consider the case for example `{'stream': ['8.1', ...]}`?

Comment (tkdchen) [inline message_tagging_service/tagging_service.py]: These two lines could be merged into one `replacements.setdefault(group, []).append(value)`

Comment (lcarva) [reply message_tagging_service/tagging_service.py]: This is needed so it can generate multiple tags. Without this, existing test cases fail.

Comment (lcarva) [reply message_tagging_service/tagging_service.py]: True! Adopted this. Thanks.

**Comment by hanzz** [comment]
+1

**Comment by tkdchen** [comment]
:+1:

**Pull request merged**